addons/test_mail/tests/test_mail_activity.py

```diff
@@ -759,3 +759,24 @@ def test_week_grouping(self):
         self.assertEqual(groups[0][groupby], pg_groups["overdue"])
         self.assertEqual(groups[1][groupby], pg_groups["today"])
         self.assertEqual(groups[2][groupby], pg_groups["planned"])
+
+@tests.tagged('mail_activity')
+class TestActivitySystray(TestActivityCommon):
+    """Test for systray_get_activities"""
+
+    def test_systray_activities_for_archived_records(self):
+        """
+            Checks if the activities made on the archived records is shown in the
+            systray activities.
+        """
+
+        test_record = self.test_record.with_user(self.user_employee)
+        test_record.action_archive()
+        test_record.activity_schedule(
+            'test_mail.mail_act_test_todo',
+            user_id=self.env.user.id,
+        )
+
+        total_count = sum(record['total_count'] for record in self.env.user.systray_get_activities() if record.get('model') == 'mail.test.activity')
```

Comment:
We could use '_name' to avoid hardcoding the model name (and made a bit of lint), like
```suggestion
        total_count = sum(
            record['total_count'] for record in self.env.user.systray_get_activities()
            if record.get('model') == test_record._name
        )
```